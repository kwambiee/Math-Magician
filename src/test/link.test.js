import React from 'react';
import TestRenderer from 'react-test-renderer';
import { BrowserRouter as Router, Link } from 'react-router-dom';

describe('renders correctly', () => {
  test('render Homepage', () => {
    const tree = TestRenderer.create(
      <Router>
        <Link to="/" className="navLink" activeStyle={{ color: '#ffd700' }}>
          <span>Home</span>
        </Link>
      </Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('render Calculator page', () => {
    const tree = TestRenderer.create(
      <Router>
        <Link
          to="/calculator"
          className="navLink"
          activeStyle={{ color: '#ffd700' }}
        >
          <span>Calculator</span>
        </Link>
      </Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('render Quote page', () => {
    const tree = TestRenderer.create(
      <Router>
        <Link
          to="/quote"
          className="navLink"
          activeStyle={{ color: '#ffd700' }}
        >
          Quotes
        </Link>
      </Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
