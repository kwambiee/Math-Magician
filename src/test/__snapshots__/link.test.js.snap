// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly render Calculator page 1`] = `
<a
  activeStyle={
    Object {
      "color": "#ffd700",
    }
  }
  className="navLink"
  href="/calculator"
  onClick={[Function]}
>
  <span>
    Calculator
  </span>
</a>
`;

exports[`renders correctly render Homepage 1`] = `
<a
  activeStyle={
    Object {
      "color": "#ffd700",
    }
  }
  className="navLink"
  href="/"
  onClick={[Function]}
>
  <span>
    Home
  </span>
</a>
`;

exports[`renders correctly render Quote page 1`] = `
<a
  activeStyle={
    Object {
      "color": "#ffd700",
    }
  }
  className="navLink"
  href="/quote"
  onClick={[Function]}
>
  Quotes
</a>
`;
